problems/200-299/problem-291.js

```diff
@@ -8,4 +8,41 @@
  * For example, given a population with weights [100, 200, 150, 80] and a boat limit of 200, the smallest
  * number of boats required will be three.
  */
-// TODO:
+const removeElement = (array, limit) => {
+    let j = array.length - 1;
+
+    while(j >= 0) {
+    if (array[j] > limit) {
+      array.splice(j, 1);
+      console.log("here", array)
+    }
+    --j;
+    }
+    return array;
+}
+
+const mainFunc = (array, limit) => {
+  let counter = 0;
+  array.sort(function(a, b){return a - b});
+  array = removeElement(array, limit);
+
+  while (array.length>0) {
+    let maxEl = array.pop();
+    array.splice(maxEl, 1);
+    let i=-1;
+    while (array[i+1]<limit-maxEl) {
+      ++i;
+      if (i>=0) {
+        array.splice(array.shift(), 1);
+      }
+    }
+    ++counter;
```

Comment:
I believe you can use a two-pointer to solve this problem. My thought is if the heaviest person can share a boat with the lightest person, then do so. Otherwise, the heaviest person can't pair with anyone, so they get their own boat.
The reason this works is that if the lightest person can pair with anyone, they might as well pair with the heaviest person.

I think it may be cleaner if you do this way. I have not tested it though. Just some thought

```
        array.sort();
        int i = 0, j = people.length - 1;
        int ans = 0;

        while (i <= j) {
            ans++;
            while (people[i] + people[j] <= limit)
                i++;
            j--;
        }

        return ans;
```

Reply:
Yes, with my version all the elements in the array get compared. This means that lightest person can share the boat with anyone. I think for this simple problem this algorithm is good.